src/providers/okta.ts

```diff
@@ -0,0 +1,104 @@
+import { OAuth2Client } from "oslo/oauth2";
+import type { OAuth2ProviderWithPKCE } from "../index.js";
+import { TimeSpan, createDate } from "oslo";
+
+export class Okta implements OAuth2ProviderWithPKCE {
+	private client: OAuth2Client;
+	private clientSecret: string;
+
+	constructor(
+		oktaDomain: string,
+		clientId: string,
+		clientSecret: string,
+		redirectURI: string,
+		options: {
+			authServerId?: string;
```

Comment:
Nitpick: Can you rename it to `authorizationServerId` since that's what Okta uses (and technically "auth" includes authentication)?

Reply:
Fixed in latest commit